src/components/search.js

```diff
@@ -10,12 +10,23 @@ import {LightTheme} from '../theme'
 import {LinkNoUnderline} from './link'
 import * as getNav from '../util/get-nav'
 import omit from '../util/omit'
+import {announce} from '../util/aria-live'
 
-const SearchResults = ({results, getItemProps, highlightedIndex}) => {
+const SearchResults = ({results, getItemProps, highlightedIndex, isOpen}) => {
   const siteMetadata = useSiteMetadata()
 
+  React.useEffect(() => {
+    if (isOpen && (!results || results.length === 0)) {
+      announce('No results')
+    }
+  }, [results, isOpen])
+
   if (!results || results.length === 0) {
-    return <Box sx={{fontSize: 2, px: 3, py: 3}}>No results</Box>
+    return (
+      <Box sx={{fontSize: 2, px: 3, py: 3}} aria-live="assertive">
+        No results
+      </Box>
+    )
```

Comment:
```suggestion
    announce('No Results')
    return (
      <Box sx={{fontSize: 2, px: 3, py: 3}} aria-live="polite">
        No Results
      </Box>
    )
```

Reply:
I'm not sure the useEffect is needed, but the announce seems to work the same way with the announce in here instead.

There is one caveat: this code is still failing to _announce_ the "No Results" on the first No Results search, but passing on the consecutive tries.

Reply:
I tried to use useEffect to resolve the first No result search but it doesn't seem to work this way. The issue is fixed as your suggested change but there is still one caveat failing on the first No result search